api/v1alpha3/azuremachinetemplate_conversion.go

```diff
@@ -54,6 +80,27 @@ func (dst *AzureMachineTemplate) ConvertFrom(srcRaw conversion.Hub) error { // n
 		return err
 	}
 
+	if src.Spec.Template.Spec.Image != nil && src.Spec.Template.Spec.Image.SharedGallery != nil {
+		if dst.Annotations == nil {
+			dst.Annotations = make(map[string]string)
+		}
+
+		// Preserve Spec.Template.Spec.Image.SharedGallery.Publisher field in annotation `azuremachinetemplate.infrastructure.cluster.x-k8s.io/sharedGallery/publisher`
+		if src.Spec.Template.Spec.Image.SharedGallery.Publisher != nil {
+			dst.Annotations[sharedGalleryPublisherAnnotation] = *src.Spec.Template.Spec.Image.SharedGallery.Publisher
+		}
+
+		// Preserve Spec.Template.Spec.Image.SharedGallery.Offer field in annotation `azuremachinetemplate.infrastructure.cluster.x-k8s.io/sharedGallery/offer`
+		if src.Spec.Template.Spec.Image.SharedGallery.Offer != nil {
+			dst.Annotations[sharedGalleryOfferAnnotation] = *src.Spec.Template.Spec.Image.SharedGallery.Offer
+		}
+
+		// Preserve Spec.Template.Spec.Image.SharedGallery.SKU field in annotation `azuremachinetemplate.infrastructure.cluster.x-k8s.io/sharedGallery/sku`
+		if src.Spec.Template.Spec.Image.SharedGallery.SKU != nil {
+			dst.Annotations[sharedGallerySKUAnnotation] = *src.Spec.Template.Spec.Image.SharedGallery.SKU
+		}
+	}
```

Comment:
do we need this? utilconversion.MarshalData should handle this, I think

Reply:
again, good call.